example/offboard_velocity/offboard_velocity.cpp

```diff
@@ -0,0 +1,229 @@
+        std::cout << ERROR_CONSOLE_TEXT << "Arming failed:" << Action::result_str(
+                      arm_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    Action::Result takeoff_result = device.action().takeoff();
+    if (takeoff_result != Action::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT << "Takeoff failed:" << Action::result_str(
+                      takeoff_result) << NORMAL_CONSOLE_TEXT << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    std::cout << "In Air" << std::endl;
+    wait_sec(5);
+
+    //////////////////////////////////////////////////////////////////////////////
+    if (offb_mode == "local-ned") {
+        // Send it once before starting offboard, otherwise it will be rejected.
+        device.offboard().set_velocity_ned({0.0f, 0.0f, 0.0f, 0.0f});
+        std::cout << "[" << offb_mode << "]" << " sent Null velocity command once before starting OFFBOARD"
+                  << std::endl;
+
+        Offboard::Result offboard_result = device.offboard().start();
+        if (offboard_result != Offboard::Result::SUCCESS) {
+            std::cout << ERROR_CONSOLE_TEXT << "Offboard start failed:" << Offboard::result_str(
+                          offboard_result) << NORMAL_CONSOLE_TEXT << std::endl;
+            return EXIT_FAILURE;
+        }
+        std::cout << "[" << offb_mode << "] OFFBOARD started" << std::endl;
+
+        // Let yaw settle.
+        std::cout << "[" << offb_mode << "] Let yaw settle..." << std::endl;
+        for (unsigned i = 0; i < 100; ++i) {
+            device.offboard().set_velocity_ned({0.0f, 0.0f, 0.0f, 90.0f});
+            wait_msec(10);
+        }
+        std::cout << "[" << offb_mode << "] Done..." << std::endl;
+        wait_sec(5);
+
+        {
+            const float step_size = 0.01f;
+            const float one_cycle = 2.0f * (float)M_PI;
+            const unsigned steps = (unsigned)(one_cycle / step_size);
+
+            std::cout << "[" << offb_mode << "]" << " Go right & oscilate" << std::endl;
+            for (unsigned i = 0; i < steps; ++i) {
+                float vx = 5.0f * sinf(i * step_size);
+                device.offboard().set_velocity_ned({vx, 0.0f, 0.0f, 90.0f});
+                wait_msec(10);
+            }
+        }
+        std::cout << "[" << offb_mode << "] Done..." << std::endl;
+        wait_sec(2);
+
+        std::cout << "[" << offb_mode << "]" << " Turn clock-wise 270 deg" << std::endl;
+        for (unsigned i = 0; i < 400; ++i) {
+            device.offboard().set_velocity_ned({0.0f, 0.0f, 0.0f, 270.0f});
+            wait_msec(10);
+        }
+        std::cout << "[" << offb_mode << "] Done..." << std::endl;
+        wait_sec(2);
+
+        std::cout << "[" << offb_mode << "]" << " Go UP 2 m/s, Turn clock-wise 180 deg" << std::endl;
+        for (unsigned i = 0; i < 400; ++i) {
+            device.offboard().set_velocity_ned({0.0f, 0.0f, -2.0f, 180.0f});
+            wait_msec(10);
```

Comment:
@julianoes I loathe waiting using timers to catch up with movement it can get unpredictable and unstable, this is something I always wanted to fix in DroneKit, we should come up with a way to either block for N time by an argument on set_velocity or find a way to have a callback

This isn't something you should worry on this PR @shakthi-prashanth-m, I'm just venting here, I would like to break this pattern in UAV development, we should be able to be more precise

Reply:
Don't know what you mean by "catch up with movement". 

Generally for velocity control examples using timers makes sense because they are something like "travel this far for X time". If we wanted to control on distance we'd use the position variants of the functions. If we do want to control on distance anyway, then we might add telemetry to measure that distance and block on meeting targets.